app/src/main/java/com/keylesspalace/tusky/view/MediaPreviewImageView.kt

```diff
@@ -15,45 +15,67 @@
 import android.support.v7.widget.AppCompatImageView
 import android.util.AttributeSet
+import com.keylesspalace.tusky.entity.Attachment
 
-import com.keylesspalace.tusky.util.FocalPointEnforcer
+import com.keylesspalace.tusky.util.FocalPointUtil
+import com.squareup.picasso.Callback
 
 /**
- * This is an extension of the standard android ImageView, which makes sure to call a focal point
- * enforcer when its size changes.
+ * This is an extension of the standard android ImageView, which makes sure to update the custom
+ * matrix when its size changes if a focal point is set.
  *
- * If the focalPointEnforcer is set on this view, it will call this enforcer to update the image
+ * If a focal point is set on this view, it will use the FocalPointUtil to update the image
  * matrix each time the size of the view is changed. This is needed to ensure that the correct
  * cropping is maintained.
  *
- * However if there is no focalPointEnforcer set (e.g. it is null), then this view should simply
+ * However if there is no focal point set (e.g. it is null), then this view should simply
  * act exactly the same as an ordinary android ImageView.
  */
 class MediaPreviewImageView
 @JvmOverloads constructor(
 context: Context,
 attrs: AttributeSet? = null,
 defStyleAttr: Int = 0
-) : AppCompatImageView(context, attrs, defStyleAttr) {
-    private var focalPointEnforcer: FocalPointEnforcer? = null
+) : AppCompatImageView(context, attrs, defStyleAttr), Callback {
+    private var focus: Attachment.Focus? = null
+    private var focalMatrix: Matrix? = null
 
     /**
-     * Set the focal point enforcer for this view.
+     * Set the focal point for this view.
      */
-    fun setFocalPointEnforcer(enforcer: FocalPointEnforcer) {
-        this.focalPointEnforcer = enforcer
+    fun setFocalPoint(focus: Attachment.Focus) {
+        scaleType = ScaleType.MATRIX
+        this.focus = focus
+        this.focalMatrix = Matrix()
     }
 
     /**
-     * Called when the size of the view changes, it calls the focalPointEnforcer to update the
-     * matrix if we have one.
+     * Called when the image is first succesfully loaded by Picasso, this function makes sure
+     * that the custom matrix of this image is initialized if a focus point is set.
+     */
+    override fun onSuccess() {
+        onSizeChanged(width, height, width, height);
+    }
+
+    // We do not handle the error here, instead it will be handled higher up the call chain.
+    override fun onError() {
+    }
+
+    /**
+     * Called when the size of the view changes, it calls the FocalPointUtil to update the
+     * matrix if we have a set focal point. It then reassigns the matrix to this imageView.
      */
     override fun onSizeChanged(width: Int, height: Int, oldWidth: Int, oldHeight: Int) {
-        if (drawable != null) {
-            focalPointEnforcer?.updateFocalPointMatrix()
+        if (drawable != null && focus is Attachment.Focus && focalMatrix is Matrix) {
```

Comment:
I would replace `is` checks with `!= null`. `is` is more about type checking

Reply:
Good point, I changed this as well :+1: